modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -303,6 +303,14 @@
   [_driver _feature _db]
   true)
 
+;; We say Mongo supports foreign keys so that the front end can use implicit
+;; joins. In reality, Mongo doesn't support foreign keys.
+;; Only define an implementation for `:foreign-keys` if none exists already.
+;; In test extensions we define an alternate implementation, and we don't want
+;; to stomp over that if it was loaded already.
+(when-not (get (methods driver/database-supports?) [:mongo :foreign-keys])
+  (defmethod driver/database-supports? [:mongo :foreign-keys] [_driver _feature _db] true))
```

Comment:
This kinda seems backwards, MongoDB really doesn't have foreign keys, I think the right way to fix this is just to remove the restriction that a database needs to support `:foreign-keys` for the FE to be able to use implicit joins. As long as someone manually marks an FK relationship and the driver supports joins they should be able to use it.

BigQuery is in the same boat so if you fix it generally then we'd be solving things for BigQuery too at the same time

Reply:
Otherwise we should just remove `:foreign-keys` altogether because it would stop meaning anything if we're enabling it even for drivers that don't have FKs